mllib/src/main/scala/org/apache/spark/mllib/rdd/MLPairRDDFunctions.scala

```diff
@@ -39,7 +39,7 @@ class MLPairRDDFunctions[K: ClassTag, V: ClassTag](self: RDD[(K, V)]) extends Se
    * @return an RDD that contains the top k values for each key
    */
   def topByKey(num: Int)(implicit ord: Ordering[V]): RDD[(K, Array[V])] = {
-    self.aggregateByKey(new BoundedPriorityQueue[V](num)(ord))(
+    self.aggregateByKey(new BoundedPriorityQueue[V](num)(ord.reverse))(
```

Comment:
I have to look closely into it tomorrow...I have been using topByKey internally and did not remember seeing this bug...